### PR TITLE
Automation of RHOSAK UseCase

### PR DESCRIPTION
**Pull request by bdattoma**
This PR wants to automate this set of Polarion IDs: ODS-248  ODS-247  ODS-246  ODS-245  ODS-243  ODS-241  ODS-239.
It follows the entire use case of RHOSAK as described in the QuickStart and as reported in the Polarion TCs:
1. enable rhosak from dashboard
2. connect to HybridCloudConsole 
3. create a Kafka stream
4. configure the stream (topic, service account and permissions)
5. run Kafka Producer and Consumer python code in JypyterLab to check if the communication with the stream is working.

The clean up code (remove rhosak, streams and configuration) is included as Test Teardown

It also brings some common keywords to JupyterLabLauncher.robot to handle different open notebook tabs.

### CODE REVIEW

**Comment by vasukulkarni** [comment]
:100:

File: tests/Resources/Page/HybridCloudConsole/Rhosak.robot
Comment (lugi0) [inline tests/Resources/Page/HybridCloudConsole/Rhosak.robot:18]: You could skip the rest of the keyword?
https://robotframework.org/robotframework/latest/libraries/BuiltIn.html#Skip

File: tests/Resources/Page/ODH/AiApps/Rhosak.robot
Comment (lugi0) [inline tests/Resources/Page/ODH/AiApps/Rhosak.robot:23]: Why not use the keywords we already have to clone from Git?

File: tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot
Comment (lugi0) [inline tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:149]: too many trailing lines

**Review by lugi0 — COMMENTED** [review]
Looks good, left a couple of comments

File: tests/Resources/Page/ODH/AiApps/Rhosak.robot
Comment (bdattoma) [reply tests/Resources/Page/ODH/AiApps/Rhosak.robot:23]: "Clone Git Repository" is one of the already developed. But actually, I think I could remove the lines from 17 to 20 and remove this "Load Kafka Notebooks From Git". 
Thanks

File: tests/Resources/Page/HybridCloudConsole/Rhosak.robot
Comment (bdattoma) [reply tests/Resources/Page/HybridCloudConsole/Rhosak.robot:18]: I will check this. My initial idea was to "skip" the rest of the TC, but then I thought that making the TC failing naturally (and giving just a warning) would implicitly test the limit on one kafka instance per user: if you get the warning message but the rest of the TC does not fail then there can be something wrong in the functionality.

What do you think?

File: tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot
Comment (bdattoma) [reply tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:149]: thanks, I will remove them

File: tests/Resources/Page/HybridCloudConsole/Rhosak.robot
Comment (lugi0) [reply tests/Resources/Page/HybridCloudConsole/Rhosak.robot:18]: Is the limit of one kafka instance per user something that we really want to test? That seems like a feature of another product, so I wouldn't want to spend time debugging issues that are not in rhods. 
I also assume this limit of one instance per user is temporary / can be increased. This might need to be revisited down the line anyways.

Comment (bdattoma) [reply tests/Resources/Page/HybridCloudConsole/Rhosak.robot:18]: makes sense yeah. But my idea is that even if this is a functionality from another product it is still faced by RHODS users. So, given that this would be implicitly tested by our TC just without skipping a line which would fail anyway, it is costless for us to test it.

Comment (tarukumar) [reply tests/Resources/Page/HybridCloudConsole/Rhosak.robot:18]: I think we can throw error instead of failure or skip it will not affect our overall test suite status also when people will go thought the logs they can see the error.We should just skip the whole block of code below to stop the failure?

Comment (bdattoma) [reply tests/Resources/Page/HybridCloudConsole/Rhosak.robot:18]: @tarukumar when you say "throw error", you mean setting the log level to ERROR or doing something like "raise ValueError" in python?

Comment (tarukumar) [reply tests/Resources/Page/HybridCloudConsole/Rhosak.robot:18]: i was suggesting to set the log level to error.

File: tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot
Comment (tarukumar) [inline tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:11]: Just a suggestion ..I think it would be better not to call individual resource file(.robot)  instead we can call the one common resource file which has all these file already present(.resource). For example : We can remove all the Jupyterhub robot file from above and just call ODHJupyterhub.resource which has all those file imported there. What do you think

Comment (lugi0) [reply tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:11]: I think this is a great suggestion and something that I never got around to doing, +1 from me

Comment (bdattoma) [reply tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:11]: agree with this, thanks!

File: tests/Resources/Page/HybridCloudConsole/Rhosak.robot
Comment (bdattoma) [reply tests/Resources/Page/HybridCloudConsole/Rhosak.robot]: I set the log level to ERROR

**Comment by bdattoma** [comment]
tested using Jenkins: [JOB#300](https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/300/TestReport/) .
Before merging, still need to apply changes based on some comments

**Comment by bdattoma** [comment]
last two testing jobs: 

- only ODS-248 run (_success_) [JOB#308](https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/308/) 

- entire PR testing suite run (_in progress_): [JOB#309](https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/309/)

File: tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot
Comment (lugi0) [inline tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:112]: This should be handled by `Spawn Notebook With Arguments` after #172 is merged

**Review by lugi0 — COMMENTED** [review]
Looks good, I would just be careful about the overlap with #172

File: tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot
Comment (bdattoma) [reply tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:112]: > Looks good, I would just be careful about the overlap with #172

I can send a new PR to fix the overlap after both are merged. Do you think the overlap may create troubles during execution?

**Review by pablofelix — APPROVED** [review]
lgtm

File: tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot
Comment (lugi0) [reply tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:112]: Yes, if both the keyword and the TC are waiting for `Wait for JupyterLab Splash Screen` the TC will fail because the keyword will give back control after the splash has already appeared

Comment (bdattoma) [reply tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:112]: you're right. I would say, if this PR is merged before the #172, i will raise a PR for correction afterwords, otherwiise I will just commit it removing those 2 lines. What do you say?

Comment (lugi0) [reply tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:112]: If it's ready to go right now just merge it and we can update it after #172

Comment (bdattoma) [reply tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:112]: yes it is ready. Just missing an approval

Comment (lugi0) [reply tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:112]: PR job for #172 is running, so up to you. I can give the approval if you want to merge now.

Comment (bdattoma) [reply tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot:112]: ok let's merge, I'll send a PR after that once 172 will be finished

**Review by lugi0 — APPROVED** [review]

**Pull request merged**
